client/app/views/TrackViewer/Viewer.vue

```diff
@@ -79,6 +92,24 @@ export default defineComponent({
       save: saveToServer, markChangesPending, pendingSaveCount,
     } = useSave(datasetId);
 
+
+    async function navigateAway() {
+      let result = true;
+      if (pendingSaveCount.value > 0) {
+        result = await prompt({
+          title: 'Save Items',
+          text: 'There is unsaved data, would you like to continue anyways?',
+          confirm: true,
+        });
+      }
+      return result;
+    }
+    async function preventNav(event: BeforeUnloadEvent) {
+      if (!pendingSaveCount.value) return;
+      event.preventDefault();
+      // eslint-disable-next-line no-param-reassign
+      event.returnValue = '';
```

Comment:
It's a recommended line for browsers to properly handle the navigation away.  It can be seen in the MDN docs as well as even the electron docs for handling reloads/closing.